include/can/core/dispatch.hpp

```diff
@@ -136,4 +136,35 @@ class Dispatcher {
     std::tuple<Listener&...> registered;
 };
 
+/**
+ * A CanMessageBufferListener that will dispatch messages to other
+ * CanMessageBufferListeners
+ * @tparam Listener CanMessageBufferListener objects
+ */
+template <CanMessageBufferListener... Listener>
+class NodeDispatcher {
```

Comment:
This is a misnomer. The point of this class is to be a `Dispatcher` that can apply a filter based on the arbitration id. The filter is just a function that returns true if the message should be passed to the `CanMessageBufferListener` or false if it should be ignored.

We discussed extending the existing `Dispatcher` class to accept the `ArbitrationIdTest` argument.  

In my opinion you either have a class that accepts a `ArbitrationIdTest` and a single `CanMessageBufferListener` OR modify `Dispatcher` to accept the `ArbitrationIdTest`. A copy and paste of the `Dispatcher` with an additional argument is overly complicated.

Reply:
changing test for ArbitrationIdTest accepting Dispatcher